src/components/Episodes/context.jsx

```diff
@@ -1,35 +1,25 @@
-import React, { createContext, useState } from "react";
-import { data } from "./";
+import React, { createContext, useReducer } from "react";
+import { 
+	reducer, 
+	data
+} from "./";
 
 const EpisodesContext = createContext();
 
 const initialState = {
 	activeIndex: 0,
-	overlayActive: false
+	overlayActive: false,
+	episodeCount: data.length
 }
 
-
 function EpisodesProvider(props){
 	const { children } = props;
 
 	const episodeCount = data.length;
-	const [activeIndex, setActiveIndex] = useState(initialState.activeIndex);
-	const [overlayActive, setOverlayActive] = useState(initialState.overlayActive);
+	const [state, dispatch] = useReducer(reducer, initialState);
 
-	const state = {
-		get: {
-			episodeCount,
-			activeIndex,
-			overlayActive
-		},
-		set: {
-			activeIndex: setActiveIndex,
-			overlayActive: setOverlayActive 
-		}
-	};
- 
 	return (
-		<EpisodesContext.Provider value={ state }>
+		<EpisodesContext.Provider value={ {state, dispatch} }>
```

Comment:
super minor - for formatting I would do either `{{ state, dispatch }}` or `{ { state, dispatch } }` here.